src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,167 @@
+                }
+                String name = dlarr[0];
+                String byStr = dlarr[1].trim();
+                LocalDateTime by;
+                try {
+                    by = LocalDateTime.parse(byStr, dtf);
+                    task = new Deadline(name, by);
+                } catch (DateTimeParseException e) {
+                    try {
+                        by = LocalDateTime.parse(byStr + currentYear, dtf);
+                        task = new Deadline(name, by);
+                    } catch (DateTimeParseException f) {
+                        try {
+                            by = LocalDateTime.parse(byStr + currentYear + DEFAULT_END, dtf);
+                            task = new Deadline(name, by);
+                        } catch (DateTimeParseException g) {
+                            throw new DukeException("Invalid time format. Please use ddMM, ddMMyy HHmm, or ddMM HHmm");
+                        }
+                    }
+
+                }
+
+                break;
+            case "e":
+            case "event":
+                String[] frarr = arr[1].split("/from ", 2);
+                if (frarr.length <= 1) {
+                    throw new DukeException("Please use the format: event /from <date/time> /to <date/time>");
+                }
+                String[] toarr = frarr[1].split("/to ", 2);
+                if (toarr.length <= 1) {
+                    throw new DukeException("Please use the format: event /from <date/time> /to <date/time>");
+                }
+                String name1 = frarr[0];
+                String fromStr = toarr[0].trim();
+                String toStr = toarr[1].trim();
+                LocalDateTime from;
+                LocalDateTime to;
+                try {
+                    from = LocalDateTime.parse(fromStr, dtf);
+                    to = LocalDateTime.parse(toStr, dtf);
+                    task = new Event(name1, from, to);
+                } catch (DateTimeParseException e) {
+                    try {
+                        from = LocalDateTime.parse(fromStr + currentYear, dtf);
+                        to = LocalDateTime.parse(toStr + currentYear, dtf);
+                        task = new Event(name1, from, to);
+                    } catch (DateTimeParseException f) {
+                        try {
+                            from = LocalDateTime.parse(fromStr + currentYear + DEFAULT_START, dtf);
+                            to = LocalDateTime.parse(toStr + currentYear + DEFAULT_END, dtf);
+                            task = new Event(name1, from, to);
+                        } catch (DateTimeParseException g) {
+                            throw new DukeException("Invalid time format. Please use ddMM, ddMMyy HHmm, or ddMM HHmm");
+                        }
+                    }
+
+                }
+
+                break;
+            default:
+                throw new DukeException("Invalid format");
```

Comment:
The case methods seems long to be, do you think it would be better to refactor to something for e.g

case "event":
     event.handler();